notebook2/notebook/notebook.py

```diff
@@ -143,7 +148,7 @@ def decorated(*args, **kwargs):
             if g.user is None:
                 if for_page:
                     session['referrer'] = request.url
-                    return redirect(external_url_for('login'))
+                    return unauthorized_redirect()
```

Comment:
Why not redirect them to the auth0 login endpoint?

Reply:
The reason is that I want to give an indication of why they're there, else it's unexpected behavior, and may annoy. I can't give that message if I send them directly to auth0 currently (there's a way to pass a message to auth0, but that's a bit more involved).

Reply:
I could redirect them to a non-generic error page, with a login button. Does that sound better?

Reply:
Actually... why don't we just try it your way. I don't want 2 clicks to get to notebook.

Reply:
One min

Reply:
Makes sense. There's already a login button on the page so I think that's ok. Might be nice to distinguish on the page whether their account is actually unauthorized or if they just need to log back in. Whatever way is easiest to convey that.